src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx

```diff
@@ -1,62 +1,83 @@
-import { ActivityIndicator, FlatList, View } from 'react-native';
-import React, { memo, useEffect, useRef, useState } from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  RefreshControl,
+  View,
+} from 'react-native';
+import React, { memo, useCallback, useRef, useState } from 'react';
 import { CommunityRepository } from '@amityco/ts-sdk-react-native';
 import { UserInterface } from '../../../types/user.interface';
 import UserItem from '../../../components/UserItem';
 import { useStyles } from '../styles';
 import { TabName } from '../../../enum/tabNameState';
+import { useFocusEffect } from '@react-navigation/native';
 
 interface ICommunityMembersTab {
   activeTab: string;
   communityId: string;
   onThreeDotTap: (user: UserInterface) => void;
   setMember: (user: UserInterface[]) => void;
+  shouldRefresh: boolean;
 }
 
 const CommunityMembersTab: React.FC<ICommunityMembersTab> = ({
   activeTab,
   communityId,
   onThreeDotTap,
   setMember,
+  shouldRefresh,
 }) => {
-  const [memberList, setMemberList] = useState<Amity.Member<'community'>[]>([]);
+  const [memberList, setMemberList] = useState<Amity.Membership<'community'>[]>(
+    []
+  );
   const [loading, setLoading] = useState(false);
-  const [hasNextPage, setHasNextPage] = useState(false);
+  const [refreshing, setRefreshing] = useState(false);
   const onNextPageRef = useRef<(() => void) | null>(null);
-  const isFetchingRef = useRef(false);
   const flatListRef = useRef(null);
   const styles = useStyles();
-
-  useEffect(() => {
+  const members = memberList.filter((member) => {
+    if (member.communityMembership === 'none') return null;
+    if (activeTab === TabName.Moderators) {
+      return member.roles.includes('community-moderator');
+    }
+    return member;
+  });
+  const getCommunityMemberList = useCallback(() => {
+    setRefreshing(true);
+    setMemberList([]);
     const option: { communityId: string; limit: number; roles?: string[] } = {
       communityId,
       limit: 10,
     };
-    option.roles =
-      activeTab === TabName.Moderators ? ['community-moderator'] : undefined;
 
-    const unsubscribeMember = CommunityRepository.Membership.getMembers(
+    const unsubscribe = CommunityRepository.Membership.getMembers(
       option,
-      ({ data: members, onNextPage, hasNextPage, loading: fetching }) => {
+      ({ data, onNextPage, hasNextPage, loading: fetching, error }) => {
         setLoading(fetching);
+        if (error) return setRefreshing(false);
         if (!fetching) {
-          setMemberList([...members]);
-          const userArray: UserInterface[] = members.map((member) => {
+          setMemberList([...data]);
```

Comment:
Why not just put `data` to `setMemberList` directly ?

Reply:
We are subscribing members from community and we need a new array for every members changed in that array. 
So I make data array as immutable to rerender the list component